scripts/paged_speculative_inference.py

```diff
@@ -141,12 +148,24 @@
 speculator = None
 if args.speculator_path is not None:
     print("loading speculator")
-    speculator = MLPSpeculator(
-        model.config.emb_dim, 4096, model.config.src_vocab_size, n_predict=3
-    )
-    speculator.load_state_dict(
-        torch.load(args.speculator_path, map_location=device)["model_state"]
-    )
+    if args.speculator_source == "fms":
+        speculator = MLPSpeculator(
+            model.config.emb_dim, 4096, model.config.src_vocab_size, n_predict=3
+        )
+        speculator.load_state_dict(
+            torch.load(args.speculator_path, map_location=device)["model_state"]
+        )
+    elif args.speculator_source == "hf":
+        from fms_extras.models.hf.modeling_mlp_speculator import (
+            MLPSpeculatorPreTrainedModel,
+        )
+
+        speculator = MLPSpeculatorPreTrainedModel.from_pretrained(
+            args.speculator_path, device_map=device
```

Comment:
in this case, I guess speculator_path is referring to a remote registered HF model name which would trigger downloading of weights?

Reply:
I fixed this to use `get_model` when the path given is local, but I think there should be a larger fix done in `fms` to handle remote weights.